src/dotnet/Telemetry.cs

```diff
@@ -9,18 +10,28 @@ namespace Microsoft.DotNet.Cli
 {
     public class Telemetry : ITelemetry
     {
+        public const double TimeoutInMilliseconds = 100;
+
         private bool _isInitialized = false;
         private TelemetryClient _client = null;
 
         private Dictionary<string, string> _commonProperties = null;
         private Dictionary<string, double> _commonMeasurements = null;
+        private Task _trackEventTask = null;
+
+        private const int ReciprocalSampleRateValue = 5;
```

Comment:
Also to address 
"Looks like as it is implemented, you won't get --version, help, info commands tracked with telemetry. Is that OK?"

This is being investigated as to how to better handle this scenario.